gnat2goto/driver/goto_utils.ads

```diff
@@ -59,6 +59,45 @@ package GOTO_Utils is
                   and then Kind (Index_Type) in Class_Type),
        Post => Kind (Compute_Memory_Op_Size'Result) = I_Op_Mul;
 
+   function Build_Function (Name : String; RType : Irep; Func_Params : Irep;
+                            FBody : Irep; A_Symbol_Table : in out Symbol_Table)
+                            return Symbol
+     with Pre => (Kind (RType) in Class_Type
+                  and then Kind (Func_Params) = I_Parameter_List
+                  and then Kind (FBody) = I_Code_Block);
+
+   function Build_Array_Size (Array_Comp : Irep; Idx_Type : Irep)
+                              return Irep
+     with Pre => ((Kind (Array_Comp) = I_Struct_Expr
+                  or else Kind (Array_Comp) = I_Symbol_Expr)
+                  and then Kind (Idx_Type) in Class_Type),
```

Comment:
Technically also a precondition that `Kind (Follow (Get_Type (Array_Comp))) = I_Struct_Type`? Might be too complicated to add it here though.

Reply:
Plus `Follow` needs to see the symbol table, I'm not sure it needs to be checked for every time we ask for size.